crates/router/src/connector/adyen/transformers.rs

```diff
@@ -289,139 +352,341 @@ impl TryFrom<&types::BrowserInformation> for AdyenBrowserInfo {
+        api::PaymentMethod::Card(ref card) => {
+            let adyen_card = AdyenCard {
+                payment_type: String::from("scheme"),
+                number: Some(card.card_number.clone()),
+                expiry_month: Some(card.card_exp_month.clone()),
+                expiry_year: Some(card.card_exp_year.clone()),
+                cvc: Some(card.card_cvc.clone()),
+            };
+            Ok(AdyenPaymentMethod::AdyenCard(adyen_card))
+        }
+        api::PaymentMethod::Wallet(ref wallet_data) => match wallet_data.issuer_name {
+            api_enums::WalletIssuer::GooglePay => {
+                let gpay_data = AdyenGPay {
+                    payment_type: String::from("googlepay"),
+                    google_pay_token: wallet_data
+                        .token
+                        .clone()
+                        .ok_or_else(utils::missing_field_err("token"))?,
+                };
+                Ok(AdyenPaymentMethod::Gpay(gpay_data))
+            }
+
+            api_enums::WalletIssuer::ApplePay => {
+                let apple_pay_data = AdyenApplePay {
+                    payment_type: String::from("applepay"),
+                    apple_pay_token: wallet_data
+                        .token
+                        .clone()
+                        .ok_or_else(utils::missing_field_err("token"))?,
+                };
+                Ok(AdyenPaymentMethod::ApplePay(apple_pay_data))
+            }
+            api_enums::WalletIssuer::Paypal => {
+                let wallet = AdyenPaypal {
+                    payment_type: String::from("paypal"),
+                };
+                Ok(AdyenPaymentMethod::AdyenPaypal(wallet))
+            }
+        },
+        api_models::payments::PaymentMethod::PayLater(ref pay_later_data) => match pay_later_data {
+            api_models::payments::PayLaterData::KlarnaRedirect { .. } => {
+                let klarna = AdyenKlarna {
+                    payment_type: String::from("klarna"),
+                };
+                Ok(AdyenPaymentMethod::AdyenKlarna(klarna))
+            }
+            api_models::payments::PayLaterData::AffirmRedirect { .. } => {
+                Ok(AdyenPaymentMethod::AdeynAffirm(AdyenAffirm {
+                    payment_type: String::from("affirm"),
+                }))
+            }
+            api_models::payments::PayLaterData::AfterpayClearpayRedirect { .. } => {
+                Ok(AdyenPaymentMethod::AfterPay(AdyenAfterPay {
+                    payment_type: String::from("afterpaytouch"),
+                }))
+            }
+            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
+        },
+        api_models::payments::PaymentMethod::BankTransfer
+        | api_models::payments::PaymentMethod::Paypal => {
+            Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into())
+        }
     }
 }
 
+fn get_card_specific_payment_data(
```

Comment:
try merging get_card_specific_payment_data , get_wallet_specific_payment_data and get_paylater_specific_payment_data into one method as there are code duplication for other params.

Reply:
@jagan-jaya the reason for keeping separate method is to increase the code readability. I understand this will increase code duplicacy (However I reduced most of duplicacy by introducing methods, from and tryfrom impl), but handling multiple conditions inside one method makes the code looks more messier.